### PR TITLE
[docs] Add a brief explanation about linters

### PR DESCRIPTION
**Pull request by uilianries**
So far, we don't a catalog with linters description and configuration file.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: docs/linters.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+## Conan Center Index Linters
+
+Some linter configuration files are available in the folder [linter](../linter), which are executed by Github Actions to improve recipe quality.
+They consume python scripts which are executed to fit CCI rules. Those scripts use `astroid` and `pylint` classes.
```
Comment (SSE4) [inline docs/linters.md:4]: links to astroid/pylint would be nice

File: docs/linters.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+To run any linter locally, read [Running the linter locally](v2_linter.md#running-the-linter-locally).
+
+
+#### [Pylint Recipe](../linter/pylintrc_recipe)
+
+This rcfile lists plugins and rules to be executed over all recipes (not test package) and validate them:
```
Comment (SSE4) [inline docs/linters.md:11]: what is rcfile?

**Review by SSE4 — DISMISSED** [review]

File: docs/linters.md (hunk 1 above)
Comment (uilianries) [reply docs/linters.md]: Done.

File: docs/linters.md (hunk 2 above)
Comment (uilianries) [reply docs/linters.md]: Done.

**Comment by uilianries** [comment]
@SSE4 thank you for your review. I've updated according your comments. Plus, I explained a bit each linter error and warning.

**Review by SSE4 — APPROVED** [review]

File: docs/linters.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+[pylint](https://pylint.pycqa.org/en/latest/) classes to parse Conan recipe files and manage their warnings and errors.
+
+The pylint by itselt is not able to find Conan Center Index rules, so astroid is used to iterate over conanfiles content and
+validate CCI conditions. Also, pylint uses [rcfile](https://pylint.pycqa.org/en/latest/user_guide/configuration/index.html)
+(a configuration file based on [toml](https://toml.io/en/) format) to configure plugins, warnings and errors which should be enabled or disabled.
+
+Also, the Github [code review](https://github.com/features/code-review) is integrated with the pylint output,
+parsed by [recipe_linter.json](../linter/recipe_linter.json), then presented to all users on the tab `Files changed`.
+
+Of course, if wanto to run any linter locally, before pushing your code, read [Running the linter locally](v2_linter.md#running-the-linter-locally).
```
Comment (prince-chrismc) [inline docs/linters.md:14]:
```suggestion
### Running Linters

Of course, you should run any linter locally, before pushing your code, read [Running the linter locally](v2_linter.md#running-the-linter-locally) for instructions.
```

File: docs/linters.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+#### [Pylint Test Package Recipe](../linter/pylintrc_testpackage)
+
+This rcfile lists plugins and rules to be executed over all recipes in test package folders only:
+
+### Linter Warning and Errors
+
+Here is the list of current warning and errors provided by pylint, when using CCI configuration.
+
+
+#### W9006 - conan-import-conanfile: ConanFile should be imported from conan
```
Comment (uilianries) [inline docs/linters.md:34]:
```suggestion
#### E9006 - conan-import-conanfile: ConanFile should be imported from conan
```
https://github.com/conan-io/conan-center-index/commit/c3f7f5af0aaa957d287e9befd7f1028bac586a0a

**Review by prince-chrismc — APPROVED** [review]

**Comment by uilianries** [comment]
Up!

**Comment by ericLemanissier** [comment]
Reposting here: The sooner this is merged the better, because right now every push to a fork of CCI starts a github action which fails, and most probably sends a notification to the pusher : https://github.com/ericLemanissier/conan-center-index/actions/workflows/marldown-links.yml

**Comment by prince-chrismc** [comment]
Please sync you branch to master so it will run the link checks 🙏

**Comment by ericLemanissier** [comment]
if you merge master into your branch or rebase on top of master, you'll need to integrate the changes of https://github.com/conan-io/conan-center-index/pull/13230 too, or CI will fail

**Comment by uilianries** [comment]
PLease, someone review and approve if good enough

**Review by jwillikers — APPROVED** [review]

**Comment by prince-chrismc** [comment]
It's okay if the markdown check is red it's not a required one https://github.com/conan-io/conan-center-index/blob/c5cab60fd26ea65e0715a4a8dd77c2dcd7d31ae5/.c3i/config_v1.yml#L26

**Comment by ericLemanissier** [comment]
OK. Sorry I did not realize my approval would have make it mergeable!

**Comment by prince-chrismc** [comment]
Thats why there's lots of us! Too much to remember for one person 🙈 

You guys are the best 💟

**Pull request merged**